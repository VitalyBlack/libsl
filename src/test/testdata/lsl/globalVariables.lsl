libsl "1.0.0";
library simple;

types {
   Int(int);
}

automaton A : Int {
   state S;

   fun f(param: Int) {
         globalInt = 1;
   }
}

var globalInt: Int = new A(state = S);
