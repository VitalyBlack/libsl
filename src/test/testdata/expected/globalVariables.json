{
  "name": "simple",
  "lslVersion": "1.0.0",
  "types": [
    {
      "name": "Int",
      "isPointer": false,
      "kind": "simple",
      "realType": {
        "name": "int",
        "isPointer": false,
        "kind": "real"
      }
    }
  ],
  "automata": [
    {
      "name": "A",
      "type": "Int",
      "constructorVariables": [],
      "variables": [],
      "states": [
        {
          "name": "S",
          "kind": "SIMPLE"
        }
      ],
      "shifts": [],
      "functions": [
        {
          "name": "f",
          "automaton": "A",
          "target": "A",
          "args": [
            {
              "name": "param",
              "type": "Int"
            }
          ],
          "contracts": [],
          "statements": [
            {
              "kind": "assignment",
              "variable": "globalInt",
              "value": {
                "kind": "integer",
                "value": 1
              }
            }
          ]
        }
      ]
    }
  ],
  "variables": [
    {
      "name": "globalInt",
      "fullName": "globalInt",
      "type": {
        "name": "Int",
        "isPointer": false,
        "kind": "simple",
        "realType": {
          "name": "int",
          "isPointer": false,
          "kind": "real"
        }
      },
      "initValue": {
        "kind": "callAutomatonConstructor",
        "automatonName": "A",
        "state": "S",
        "args": []
      },
      "kind": "global"
    }
  ]
}